core-processor/src/ext.rs

```diff
@@ -802,7 +818,7 @@ impl EnvExt for Ext {
     }
 
     fn wake(&mut self, waker_id: MessageId, delay: u32) -> Result<(), Self::Error> {
-        self.charge_gas(self.context.message_context.settings().waking_fee())?;
+        self.charge_gas_if_enough(self.context.message_context.settings().waking_fee())?;
```

Comment:
I don't like this naming. Can we go with something like `charge_gas_checked` maybe?

Reply:
Frankly speaking the old one seemed better

Reply:
Agree with gshep